wireapp/wire-android
# fix: missing remove device button [WPB-8819]

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-8819" title="WPB-8819" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />WPB-8819</a>  [Android] "Remove Device" button not showing in Beta app
  </td></table>
  <br />
 






















----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

Sometimes "remove device" button is missing for some old clients.

### Causes (Optional)

The `clientType` field could be null.

### Solutions

Use kalium fix to update this field properly on conflict.
Hide "remove device" button only for legal hold devices, so in case this field is null, button will be visible.
Fix loading states on `SelfDevicesScreen` and `RemoveDeviceScreen`.
Also, add handling errors on these screens when getting and fetching clients list, if some crucial data are missing then show dialog error with the option to retry.
Update `DeviceItem` UI element to match designs.

### Dependencies (Optional)

Needs releases with:

- https://github.com/wireapp/kalium/pull/2821

### Testing

#### Test Coverage (Optional)


#### How to Test

Open "my devices" screen or "remove device" screen (must reach the limit of registered clients).
To verify whether "remove device" button is not missing, you probably need to have the app version from before https://github.com/wireapp/kalium/pull/1544 and update it.

### Attachments (Optional)

| Before | After |
| ----------- | ------------ |
| <img width="400" alt="before" src="https://github.com/wireapp/wire-android/assets/30429749/5a1423e4-db57-4746-84ac-5151c707868c"> | <img width="400" alt="after" src="https://github.com/wireapp/wire-android/assets/30429749/703e2ae7-b62b-4d30-8860-603cb0a5478e"> |
|   | <img width ="400" src="https://github.com/wireapp/wire-android/assets/30429749/a5cb5380-654d-4e01-8ce6-9b93da9c69ba"/> |
| <video width="400" src="https://github.com/wireapp/wire-android/assets/30429749/309bcdf1-a039-4ef4-9143-2bb6e9f47723"/> | <video width="400" src="https://github.com/wireapp/wire-android/assets/30429749/2b9f09dc-3a5f-4e00-bc7f-d6a8131b7485"/> |

Note: for these videos, a delay of 3s was added to show the difference in loading state.

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
